gtsfm/data_association/point3d_initializer.py

```diff
@@ -192,13 +205,46 @@ def triangulate(self, track_2d: SfmTrack2d) -> Tuple[Optional[SfmTrack], Optiona
 
         # all the measurements should have error < threshold
         if not np.all(reproj_errors < self.reproj_error_thresh):
-            return None, avg_track_reproj_error, is_cheirality_failure
+            return None, avg_track_reproj_error, TriangulationResult.HIGH_REPROJECTION_ERROR
 
         track_3d = SfmTrack(triangulated_pt)
         for i, uv in inlier_track.measurements:
             track_3d.add_measurement(i, uv)
 
-        return track_3d, avg_track_reproj_error, is_cheirality_failure
+        return track_3d, avg_track_reproj_error, TriangulationResult.SUCCESS
+
+    def is_small_triangulation_angle(self, point_3d: np.ndarray, camera_indices: List[int]) -> bool:
+        """Check if all pairs of cameras have a small triangulation angle.
+
+        References:
+        - https://github.com/colmap/colmap/blob/513fef3e39f99b50279bf5aee1c5b597cba53c3b/src/base/reconstruction.cc#L616
+        - https://github.com/sweeneychris/TheiaSfM/blob/d2112f15aa69a53dda68fe6c4bd4b0f1e2fe915b/src/theia/sfm/set_outlier_tracks_to_unestimated.cc#L121
+
+
+        Args:
+            point_3d: the 3d point which has been triangulated.
+            camera_indices: indices of the cameras associated with the triangulation.
+
+        Returns:
+            True if all the pairs of camera subtend a small angle, and hence resulting in a small baseline track.
+        """
+
+        if self.min_baseline_thresh is None:
+            raise TypeError("The threshold should not be None for the check")
+
+        for i1, i2 in itertools.combinations(camera_indices, 2):
+            camera_i1 = self.track_camera_dict[i1]
+            camera_i2 = self.track_camera_dict[i2]
+
+            triangulation_angle = triangulation_utils.calculate_triangulation_angle_in_degrees(
+                camera_i1, camera_i2, point_3d
+            )
+
+            if triangulation_angle > self.min_baseline_thresh:
```

Comment:
so our logic here is that if any pair has a low triangulation error, then we discard the track.

but doesnt COLMAP instead say that if any camera pair has a large enough triangulation angle, then the whole track is kept?

i would probably prefer COLMAP's metric.

Reply:
My logic is the same too. if I have an angle which is greater than the min threshold, I return false for the *small triangulation angle* check. So I am essentially doing the same thing.